Automattic/woocommerce-payments
# UPE styles for focused and invalid inputs.

Fixes #1913 

#### Changes proposed in this Pull Request

Compute styles for invalid inputs and focused inputs.
As the styles are not present on the page to be computed, we add a hidden div and append one invalid and one focused input.
After computing the styles, we remove the hidden container.









#### Testing instructions

* Enter your WC Options table and add the feature flag _wcpay_feature_upe with a value of 1. Stripe UPE payment gateway should now replace the Credit Card and be enabled at checkout.
* Please go to Payments > Settings and ensure that WooCommerce Payments is enabled.
* Add an item from your shop to your cart and proceed through a regular checkout. At checkout, select "WooCommerce Payments" and you should see the UPE appear below the payment gateway.
* Click on one of the inputs of the UPE element, check the styles are similar as when you click on one of the inputs in the WooCommerce form. Pay attention to the outline and color of the text.
* Enter an invalid information, like '11' on CVC input, check the styles correspond to invalid inputs on the WooCommerce form.

A few screenshots:
<img width="411" alt="StorefrontInvalid" src="https://user-images.githubusercontent.com/8002881/120024312-96cd7d80-bfb4-11eb-8ffb-58d1d2cecd0e.png">
<img width="408" alt="StorefrontFocus" src="https://user-images.githubusercontent.com/8002881/120024315-99c86e00-bfb4-11eb-87d0-618aba720685.png">
<img width="513" alt="AstraFocusDate" src="https://user-images.githubusercontent.com/8002881/120024320-9cc35e80-bfb4-11eb-97a7-4a1d18fcbfc4.png">
<img width="519" alt="AstraInvalid" src="https://user-images.githubusercontent.com/8002881/120024328-9e8d2200-bfb4-11eb-8482-ffffa5bd8ba5.png">
<img width="772" alt="2021Focus" src="https://user-images.githubusercontent.com/8002881/120025069-bb762500-bfb5-11eb-9238-8262247894e7.png">





*

-------------------

- [x] Tested on mobile (or does not apply) ->  Tested but not comprehensively.
